src/main/java/org/terasology/spawning/SpawningAuthoritySystem.java

```diff
@@ -0,0 +1,171 @@
+        }
+    }
+
+    public boolean consumeItem(EntityRef player, Prefab prefab) {
+        OreonSpawnComponent oreonSpawnComponent = prefab.getComponent(OreonSpawnComponent.class);
+        HashMap<String, Integer> itemsHashMap = oreonSpawnComponent.itemsToConsume;
+
+        if (oreonSpawnComponent.itemsToConsume != null) {
+            ArrayList<String> neededItemList = new ArrayList(itemsHashMap.keySet());
+            int numberOfItems = neededItemList.size();
+            if (player.hasComponent(InventoryComponent.class)) {
+                ArrayList<Integer> requiredSlots = getSlotsForRequiredItems(itemsHashMap, player);
+
+                //all required items not in inventory
+                if (requiredSlots.size() != numberOfItems) {
+                    logger.info("Could not find all required items in inventory");
+                    return false;
+                }
+
+                for (int slotNumber = 0; slotNumber < numberOfItems; slotNumber++) {
+                    if(!removeNeededItem(itemsHashMap, requiredSlots.get(slotNumber), player)) {
+                        //could not decrement the required number from inventory
+                        return false;
+                    }
+                }
+
+                logger.info("Found all items required for spawning, creating your Oreon");
+                return true;
+            }
+
+            logger.info("No inventory to source material from, cannot spawn");
+            return false;
+        }
+
+        logger.info("No item required to spawn, spawning your Oreon");
+        return true;
+    }
+
+    public ArrayList<Integer> getSlotsForRequiredItems(HashMap<String, Integer> items, EntityRef player) {
+        ArrayList<Integer> requiredSlots = new ArrayList<>();
+
+        int inventorySize = inventoryManager.getNumSlots(player);
+
+        for(int slotNumber = 0; slotNumber <= inventorySize; slotNumber++) {
+            EntityRef inventorySlot = inventoryManager.getItemInSlot(player, slotNumber);
+
+            BlockItemComponent blockItemComponent = inventorySlot.getComponent(BlockItemComponent.class);
+            if (blockItemComponent != null) {
+                String blockFamily = blockItemComponent.blockFamily.toString();
+                //if this item is required
+                if (items.containsKey(blockFamily)) {
+                    requiredSlots.add(slotNumber);
+                }
+            }
+        }
+
+        return requiredSlots;
+    }
+
+    public boolean removeNeededItem(HashMap<String, Integer> items, int slotNumber, EntityRef player) {
+        EntityRef inventorySlot = inventoryManager.getItemInSlot(player, slotNumber);
+        BlockItemComponent blockItemComponent = inventorySlot.getComponent(BlockItemComponent.class);
+        String blockFamily = blockItemComponent.blockFamily.toString();
```

Comment:
blockFamilyName is probably better than blockFamily.   You have an actual blockFamily object on the same line.